From a3fcec980d2bfe727b8c1b00b493840d198f0e7c Mon Sep 17 00:00:00 2001
From: Kenny Kessler <kenny.kessler@motorola.com>
Date: Wed, 21 Oct 2015 15:41:32 -0500
Subject: [PATCH 414/959] IKSWM-8062: power: qpnp-smbcharger-mmi: Disable LED

SMBLITE versisons support LED control.  Disable the
LED by setting it to SW control with no Blink.

Change-Id: I2bb321a31ed80aeb40ee7acbd47715c96420dd7d
Signed-off-by: Kenny Kessler <kenny.kessler@motorola.com>
Reviewed-on: http://gerrit.mot.com/793381
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Jason Knopsnyder <jasonk@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/power/qpnp-smbcharger-mmi.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index d6d1f378272..4f64dc1ba79 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -5264,11 +5264,26 @@ static inline int get_bpd(const char *name)
 #define USBIN_ALLOW_5V_UNREG		0x4
 #define USBIN_ALLOW_5V_9V_UNREG		0x5
 #define DCIN_ALLOW_5V_TO_9V		0x2
+#define CMD_CHG_LED_REG			0x43
+#define CMD_CHG_LED_BLINK_MASK		SMB_MASK(2, 1)
+#define CMD_CHG_LED_CTRL_BIT		BIT(0)
 static int smbchg_hw_init(struct smbchg_chip *chip)
 {
 	int rc, i;
 	u8 reg, mask;
 
+	/* Disable Charge LED on PMI8950 Only */
+	if (chip->schg_version == QPNP_SCHG_LITE) {
+		rc = smbchg_masked_write(chip,
+					 chip->bat_if_base + CMD_CHG_LED_REG,
+					 CMD_CHG_LED_BLINK_MASK |
+					 CMD_CHG_LED_CTRL_BIT,
+					 CMD_CHG_LED_CTRL_BIT);
+		if (rc < 0)
+			dev_err(chip->dev, "Couldn't disable LED rc=%d\n",
+				rc);
+	}
+
 	rc = smbchg_read(chip, chip->revision,
 			chip->misc_base + REVISION1_REG, 4);
 	if (rc < 0) {
-- 
2.11.0

